File: GGAndroid_2.0/GGCore/GGVMCore/src/main/java/io/ganguo/vmodel/view/AdapterInterface.java
package io.ganguo.vmodel.view;

import android.databinding.ViewDataBinding;

import io.ganguo.library.ui.adapter.v7.hodler.BaseViewHolder;
import io.ganguo.library.ui.view.ViewInterface;
import io.ganguo.vmodel.adapter.ViewModelAdapter;

/**
 * Created by Roger on 6/7/16.
 */
public interface AdapterInterface<T extends ViewDataBinding> extends ViewInterface<T> {
    ViewModelAdapter<T> getAdapter();

    BaseViewHolder<T> getViewHolder();
}

File: GGAndroid_3.0/core/frame/mvvm-v1/src/main/java/io/ganguo/viewmodel/core/viewinterface/AdapterInterface.kt
package io.ganguo.viewmodel.core.viewinterface

import androidx.databinding.ViewDataBinding
import io.ganguo.adapter.hodler.ItemViewHolder
import io.ganguo.viewmodel.core.adapter.ViewModelAdapter


/**
 * <pre>
 *     author : leo
 *     time   : 2019/10/11
 *     desc   : RecyclerView Adapter Interface
 * </pre>
 */
interface AdapterInterface<T : ViewDataBinding> : ViewInterface<T> {
    val adapter: ViewModelAdapter
    val viewHolder: ItemViewHolder<T>
}